tutorials/navigation/navigation_using_navigationmeshes.rst

```diff
@@ -17,254 +19,324 @@ Navigation meshes are the only things considered when doing pathfinding, e.g. vi
+
+.. warning::
+
+    The data from visual meshes and polygons need to be received from the GPU, stalling the RenderingServer in the process.
+    For runtime (re)baking prefer using physics shapes as parsed source geometry.
 
-    var new_2d_region_rid: RID = NavigationServer2D.region_create()
+Source geometry is stored inside resources so the created geometry can be reused for multiple bakes.
+E.g. baking multiple navigation meshes for different agent sizes from the same source geometry.
+This also allows to save source geometry to disk so it can be loaded later, e.g. to avoid the overhead of parsing it again at runtime.
 
-    var default_2d_map_rid: RID = get_world_2d().get_navigation_map()
-    NavigationServer2D.region_set_map(new_2d_region_rid, default_2d_map_rid)
+The geometry data should be in general kept very simple. As many edges as are required but as few as possible.
+Especially in 2D duplicated and nested geometry should be avoided as it forces polygon hole calculation that can result in flipped polygons.
+An example for nested geometry would be a smaller StaticBody2D shape placed completely inside the bounds of another StaticBody2D shape.
 
-    var new_navigation_polygon: NavigationPolygon = NavigationPolygon.new()
-    var new_outline: PackedVector2Array = PackedVector2Array([
-        Vector2(0.0, 0.0),
-        Vector2(50.0, 0.0),
-        Vector2(50.0, 50.0),
-        Vector2(0.0, 50.0),
-    ])
-    new_navigation_polygon.add_outline(new_outline)
-    new_navigation_polygon.make_polygons_from_outlines()
+Navigation mesh baking common problems
+--------------------------------------
 
-    NavigationServer2D.region_set_navigation_polygon(new_2d_region_rid, new_navigation_polygon)
+There are some common user problems and important caveats to consider when creating or baking navigation meshes.
 
-Procedural 3D navigation mesh
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Navigation mesh baking creates frame rate problems at runtime
+    The navigation mesh baking is by default done on a background thread so as long as the platform supports threads the actual baking is rarely the source of any performance issues (assuming a reasonably sized and complex geometry for runtime rebakes).
 
-The following script creates a new 3D navigation region and fills it with procedurally generated navigation mesh data from a NavigationMesh resource.
+    The common source for performance issues at runtime is the parsing step for source geometry that involves nodes and the SceneTree. The SceneTree is not thread-safe so all the nodes need to be parsed on the main-thread. Some nodes with a lot of data can be very heavy and slow to parse at runtime, e.g. a TileMap has one or more polygons for every single used cell and TileMapLayer to parse. Nodes that hold meshes need to request the data from the RenderingServer stalling the rendering in the process.
```

Comment:
```suggestion
    The common source for performance issues at runtime is the parsing step for source geometry that involves nodes and the SceneTree.
    The SceneTree is not thread-safe so all the nodes need to be parsed on the main-thread.
    Some nodes with a lot of data can be very heavy and slow to parse at runtime, e.g. a TileMap has one or more polygons for every single used cell and TileMapLayer to parse.
    Nodes that hold meshes need to request the data from the RenderingServer stalling the rendering in the process.
```
These big blocks need to be broken up for the source, the lines should not be too wide, compare with for example lines 178-181